libkbfs/block_types.go

```diff
@@ -188,10 +188,30 @@ func (fb *FileBlock) NewEmpty() Block {
 
 // DataVersion returns data version for this block.
 func (fb *FileBlock) DataVersion() DataVer {
+	// If this is an indirect block, and none of its children are
+	// marked as direct blocks, then this must be a big file.  Note
+	// that we do it this way, rather than returning on the first
+	// non-direct block, to support appending to existing files and
```

Comment:
If we're modifying an old (v2) indirect block, shouldn't we set the `DirectType` on all its indirect pointers so that we can ensure that most indirect blocks will have that field set?

In that case, can't we then return on the first indirect block? I.e. something like this:

```
if fb.IPtrs is empty, return V1

if any IPtr in fb.IPtrs has unknown direct type, panic
if there are multiple values of IPtr.DirectType, panic

If the single DirectType is indirect, return V3
else if any IPtr has holes, return V2
else, return V1
```